ipam/allocate.go

```diff
@@ -23,23 +24,34 @@ func (g *allocate) Try(alloc *Allocator) bool {
 		return true
 	}
 
-	// If we have previously stored an address for this container, return it.
-	if addr, found := alloc.owned[g.ident]; found {
-		g.resultChan <- allocateResult{addr, nil}
+	// If we have previously stored an address for this container in this subnet, return it.
+	if addrs, found := alloc.owned[g.ident]; found {
+		for _, addr := range addrs {
+			if g.subnet.Contains(addr) {
+				g.resultChan <- allocateResult{addr, nil}
+				return true
+			}
+		}
+	}
+
+	if !alloc.universe.Overlaps(g.subnet) {
```

Comment:
Overlapped seemed more broadly useful than contained, as you might want both IPAM-allocated and manually-allocated addresses in the same subnet.  I.e. you could give IPAM 10.3.0.0/16 but run all containers in 10.0.0.0/8.
I agree that this way round is harder to grasp, initially.